onnx/defs/shape_inference.h

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+#include "data_type_utils.h"
+
+namespace ONNX_NAMESPACE {
+
+struct InferenceContext {
+  virtual const AttributeProto* getAttribute(const std::string& name) const = 0;
+  virtual size_t getNumInputTypes() const = 0;
```

Comment:
actually rename it to numInputs/getNumInputs? "Types" part is confusing

Reply:
sure